src/core/Form/FilterInput/FilterInput.test.tsx

```diff
@@ -0,0 +1,199 @@
+        'aria-describedby',
+        '123-statusText',
+      );
+    });
+  });
+
+  describe('disabled', () => {
+    it('has disabled attribute and classname', () => {
+      const { container, getByRole } = render(
+        <FilterInput
+          disabled={true}
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      expect(container.firstChild).toHaveClass('fi-filter-input--disabled');
+
+      const inputField = getByRole('textbox') as HTMLInputElement;
+      expect(inputField).toHaveAttribute('disabled');
+    });
+  });
+
+  describe('labelText', () => {
+    it('should be found ', () => {
+      const { getByText } = render(
+        <FilterInput
+          labelText="Label"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const label = getByText('Label');
+      expect(label).toHaveClass('fi-label-text_label-span');
+    });
+  });
+
+  describe('optionalText', () => {
+    it('should have element and correct classname for it', () => {
+      const { getByText } = render(
+        <FilterInput
+          labelText="Label"
+          optionalText="Optional"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const optionalText = getByText('(Optional)');
+      expect(optionalText).toHaveClass('fi-label-text_optionalText');
+    });
+  });
+
+  describe('labelMode', () => {
+    it('should be visible by default', () => {
+      const { getByText } = render(
+        <FilterInput
+          labelText="Label"
+          optionalText="Optional"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const label = getByText('Label');
+      expect(label).toHaveClass('fi-label-text_label-span');
+    });
+
+    it('should be hidden', () => {
+      const { getByText } = render(
+        <FilterInput
+          labelText="Label"
+          labelMode="hidden"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const label = getByText('Label');
+      expect(label).toHaveClass('fi-visually-hidden');
+    });
+  });
+
+  describe('visualPlaceholder', () => {
+    it('should have the given text', () => {
+      const { getByRole } = render(
+        <FilterInput
+          labelText="Label"
+          visualPlaceholder="Enter text here"
+          items={tools}
+          onFiltering={(filtered) => console.log(filtered)}
+          filterRule={filter}
+        />,
+      );
+      const inputField = getByRole('textbox') as HTMLInputElement;
+      expect(inputField).toHaveAttribute('placeholder', 'Enter text here');
+    });
+  });
+});
```

Comment:
Currently there are no tests for following props: `items`, `onFiltering` and` filterRule`.

Reply:
Looks like there is now a test for all of the above. Good to resolve this in my opinion. Or at least the mock.results test looks like it.

Reply:
Yes, these do now have tests.